client.go

```diff
@@ -7,22 +7,45 @@ import (
 
 // Client holds client configuration
 type Client struct {
-	baseURL       string
-	authorization string
+	baseURL         string
+	authorization   string
+	customUserAgent string
 }
 
 const defaultBaseURL = "https://api.fivetran.com/v1"
 const restAPIv2 = "application/json;version=2"
 
+// WARNING: Update Agent version on each release!
+const defaultUserAgent = "Go-Fivetran/0.2.3"
+
 // New receives API Key and API Secret, and returns a new Client
 func New(apiKey, apiSecret string) *Client {
 	return &Client{
-		baseURL:       defaultBaseURL,
-		authorization: fmt.Sprintf("Basic %v", base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%v:%v", apiKey, apiSecret)))),
+		baseURL:         defaultBaseURL,
+		authorization:   fmt.Sprintf("Basic %v", base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%v:%v", apiKey, apiSecret)))),
+		customUserAgent: "",
 	}
 }
 
 // BaseURL changes Client base REST API endpoint URL
 func (c *Client) BaseURL(baseURL string) {
 	c.baseURL = baseURL
 }
+
+// UserAgent sets custom User-Agent header in Client requests
+func (c *Client) CustomUserAgent(customUserAgent string) {
+	c.customUserAgent = customUserAgent
+}
+
+func (c *Client) commonHeaders() map[string]string {
+	userAgent := defaultUserAgent
+
+	if c.customUserAgent != "" {
+		userAgent += " " + c.customUserAgent
+	}
+
```

Comment:
I think the map needs to be initialized first. 
```go
commonHeaders := make(map[string]string)
...
return commonHeaders
}
```
```

Reply:
@shirolimit proposed the opposit approach, initially I've done this with make